suite-native/device/src/hooks/useWipeDevice.tsx

```diff
@@ -0,0 +1,47 @@
+import { useCallback, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { useNavigation } from '@react-navigation/native';
+
+import { selectSelectedDevice } from '@suite-common/wallet-core';
+import { requestPrioritizedDeviceAccess } from '@suite-native/device-mutex';
+import {
+    DeviceOnboardingStackRoutes,
+    RootStackParamList,
+    RootStackRoutes,
+    StackNavigationProps,
+} from '@suite-native/navigation';
+import TrezorConnect from '@trezor/connect';
+
+type NavigationProp = StackNavigationProps<RootStackParamList, RootStackRoutes.BackupFailedModal>;
+
+export const useWipeDevice = () => {
+    const [isWipeInProgress, setIsWipeInProgress] = useState(false);
+    const navigation = useNavigation<NavigationProp>();
+    const device = useSelector(selectSelectedDevice);
+
+    const wipeDevice = useCallback(async () => {
+        setIsWipeInProgress(true);
+        if (!device) return;
+
+        const response = await requestPrioritizedDeviceAccess({
+            deviceCallback: async () =>
+                await TrezorConnect.wipeDevice({
+                    device: {
+                        path: device.path,
+                    },
+                    // In bootloader mode we need the skip the final reload otherwise we never get the resolution
+                    skipFinalReload: device.mode === 'bootloader',
+                }),
+        });
+
+        if (response.success && response.payload.success) {
+            navigation.navigate(RootStackRoutes.DeviceOnboardingStack, {
```

Comment:
should not be the user redirected to the HomeScreen? https://www.figma.com/design/xSWRW8Csq7zYJ3TiqwoIxB/Small-Things?node-id=3130-13415&t=zeJW2o5AK3f3b8OI-1. I think the question "Have you used this Trezor before" sounds a bit redundant

Reply:
This is all part of the next issue. This one is only displaying of the tiles and functionality with navigation is coming as a follow up in https://github.com/trezor/trezor-suite/issues/18913